### ISSUE TITLE
@nx/angular:ng-packagr-lite fails in ci

### ISSUE THREAD

**Issue by Squixx** [nrwl/nx#22454]
### Current Behavior

Some of our libraries using `@nx/angular:ng-packagr-lite`  (silently) fail to build in Ci when using NX >= 18.1, but build perfectly fine when directly using ` @angular-devkit/build-angular:ng-packagr` 

we're running on nx cloud on medium agents.

### Expected Behavior

Build succeeds or throws a usefull error

### Steps to Reproduce

Build a relatively large library using either @nx/angular:ng-packagr-lite or @nx/angular:package on a low-end machine or a CI environment. Unfortunately, this issue arises in all versions newer than 18.0.8, possibly due to the switch to Angular 17.2 CLI.

We suspect a memory leak might be the cause, as directly calling the Angular executor works fine. However, using the executor directly is not feasible for us as a workaround because we would miss out on some options.

Currently, I've tested this with the latest canary release of NX, but the issue persists on any version beyond 18.0.8.

### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node   : 20.11.1
OS     : linux-x64
pnpm   : 8.15.5

nx (global)        : 18.0.3
nx                 : 18.2.0-canary.20240321-2a4c57d
@nx/js             : 18.2.0-canary.20240321-2a4c57d
@nx/jest           : 18.2.0-canary.20240321-2a4c57d
@nx/eslint         : 18.2.0-canary.20240321-2a4c57d
@nx/workspace      : 18.2.0-canary.20240321-2a4c57d
@nx/angular        : 18.2.0-canary.20240321-2a4c57d
@nx/cypress        : 18.2.0-canary.20240321-2a4c57d
@nx/devkit         : 18.2.0-canary.20240321-2a4c57d
@nx/eslint-plugin  : 18.2.0-canary.20240321-2a4c57d
@nx/express        : 18.2.0-canary.20240321-2a4c57d
@nx/node           : 18.2.0-canary.20240321-2a4c57d
@nx/plugin         : 18.2.0-canary.20240321-2a4c57d
@nx/rollup         : 18.2.0-canary.20240321-2a4c57d
@nx/storybook      : 18.2.0-canary.20240321-2a4c57d
@nrwl/tao          : 18.0.3
@nx/web            : 18.2.0-canary.20240321-2a4c57d
@nx/webpack        : 18.2.0-canary.20240321-2a4c57d
typescript         : 5.4.3
---------------------------------------
Community plugins:
@compodoc/compodoc       : 1.1.23
@ionic/angular           : 7.7.3
@ionic/angular-toolkit   : 11.0.1
@ngrx/effects            : 17.1.1
@ngrx/router-store       : 17.1.1
@ngrx/store              : 17.1.1
@ngrx/store-devtools     : 17.1.1
@nx-dotnet/core          : 2.2.0
@nxext/capacitor         : 17.0.1
@nxext/ionic-angular     : 17.0.1
@storybook/angular       : 7.6.17
angular-auth-oidc-client : 17.0.0
ng-mocks                 : 14.12.1
ng2-charts               : 6.0.0
ngx-toastr               : 18.0.0
---------------------------------------
Local workspace plugins:
         @company/generators
---------------------------------------
The following packages should match the installed version of nx
  - @nrwl/tao@18.0.3

To fix this, run `nx migrate nx@18.2.0-canary.20240321-2a4c57d`
```

### Failure Logs

```shell
15042   NX   Running 1 tasks:
15043  
15044  - Task companyc:build 
15045  
15046  
15047  [Nx Cloud] Unable to detect a VCS context from the environment.
15048  
15049   NX   RunStart
15050  
15051  
15052  {
15053    "meta": {},
15054    "branch": "154054",
15055    "runGroup": "1213957",
15056    "ciExecutionId": "1213957",
15057    "ciExecutionEnv": "",
15058    "distributedExecutionId": "fJIylNLwki",
15059    "hashes": [
15060      "REDACTED"
15061    ],
15062    "machineInfo": {
15063      "machineId": "2u4vi4FoQiacDpL7skJYPw==",
15064      "platform": "linux",
15065      "version": "#1 SMP Sat Dec 30 13:01:38 UTC 2023",
15066      "cpuCores": 32
15067    },
15068    "vcsContext": null
15069  }
15070  
15071  RunStart duration: 39
15072  
15073   NX   Nx Cloud: Cache miss 14570467693681281866.
15074  
15075  
15076  > nx run companyc:build
15077  
15078  companyc: Building Angular Package
15079  companyc: 
15080  companyc: ------------------------------------------------------------------------------
15081  companyc: Building entry point '@company/companyc/types'
15082  companyc: ------------------------------------------------------------------------------
15083  companyc: - Compiling with Angular sources in Ivy full compilation mode.
15084  companyc: ✔ Compiling with Angular sources in Ivy full compilation mode.
15085  companyc: Built @company/companyc/types
15086  companyc: 
15087  companyc: ------------------------------------------------------------------------------
15088  companyc: Building entry point '@company/companyc'
15089  companyc: ------------------------------------------------------------------------------
15090  companyc: - Compiling with Angular sources in Ivy full compilation mode.
15091  
15092   NX   Nx Cloud: Stored REDACTED
15093  
15094  
15095  
15096  
15097   NX   Ran 1 tasks:
15098  
15099  - Task companyc:build   FAILED
15100  
15101  
15102   NX   Executed tasks with hashes: [{"taskId":"companyc:build","hash":"REDACTED","code":1}]
15103  
15104  [Nx Cloud] Unable to detect a VCS context from the environment.
15105  
15106   NX   RunEnd. Completed tasks
15107  
15108  
15109  [
15110    {
15111      "taskId": "companyc:build",
15112      "startTime": "2024-03-22T14:40:31.611Z",
15113      "target": "build",
15114      "projectName": "companyc",
15115      "hash": "REDACTED",
15116      "hashDetails": {
15117        "nodes": {
15118          "companyc:{projectRoot}/**/*,!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap),!{projectRoot}/tsconfig.spec.json,!{projectRoot}/jest.config.[jt]s,!{projectRoot}/.eslintrc.json,!{projectRoot}/.storybook/**/*,!{projectRoot}/**/*.stories.@(js|jsx|ts|tsx|mdx),!{projectRoot}/src/test-setup.[jt]s,!{projectRoot}/tsconfig.storybook.json": "6819112670917268557",
15119          "jest-config:ProjectConfiguration": "7946772251608158350",
15120          "{workspaceRoot}/.nxignore": "7293427878140144706",
15121          "{workspaceRoot}/.nx/workflows/agents.yaml": "1209634257041243247",
15122          "companyc:TsConfig": "8292976837670921435",
15123          "jest-config:TsConfig": "5097062501614189270",
15124          "{workspaceRoot}/libs/ui-fundamentals/styles/*": "3155628918323077164",
15125          "{workspaceRoot}/nx.json": "13980571880781538030",
15126          "companyc:ProjectConfiguration": "13044197443065204185",
15127          "jest-config:{projectRoot}/**/*,!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap),!{projectRoot}/tsconfig.spec.json,!{projectRoot}/jest.config.[jt]s,!{projectRoot}/.eslintrc.json,!{projectRoot}/.storybook/**/*,!{projectRoot}/**/*.stories.@(js|jsx|ts|tsx|mdx),!{projectRoot}/src/test-setup.[jt]s,!{projectRoot}/tsconfig.storybook.json": "9398553357821906042",
15128          "{workspaceRoot}/.gitignore": "4694926061998298005",
15129          "npmDependencies": "XGuDgsBFVXGw9YNLyLmp/w=="
15130        },
15131        "runtime": {},
15132        "implicitDeps": {}
15133      },
15134      "params": "",
15135      "uploadedToStorage": true,
15136      "endTime": "2024-03-22T14:41:08.897Z",
15137      "status": 1,
15138      "cacheStatus": "cache-miss",
15139      "terminalOutput": "\u001b[34mBuilding Angula...",
15140      "artifactId": "963bb898-0894-4845-af7a-b80f68e24292"
15141    }
15142  ]
15143  
15144  RunEnd duration: 32
```

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

edit: 

Switching to build agents with higher memory bypasses the issue, so it really seems that the memory usage has become an issue after 18.0.8

**Comment by Squixx** [comment]
possibly related to https://github.com/nrwl/nx/pull/20556/commits/d30f2aa06102395429d058eb93fd98fb22791b43#diff-e3804de36ed0f9233b5d7b0bbe27c9510cb1b405c221b058f92dddbb38f00971

where the entrypoints in `complie-ngc.transform.ts` are moved outside of the try-catch block?

**Comment by leosvelperez** [comment]
Thanks for reporting this!

I'm having trouble reproducing the issue, but I think I know what the issue is. This issue was introduced by v17.2.0 of `ng-packagr` but it seems it was addressed in `ng-packagr` 17.2.1. We synced the fix back to our executors, but we wrongly made the change to apply only for Angular versions greater than 17.3.0, so that's why you're probably still getting the issue with the Nx executor while the `@angular-devkit/build-angular:ng-packagr` builder works fine.

Just to be sure, could you confirm you're using `ng-packagr` v17.2.1 or greater? Since I can't reproduce the issue, that would allow me to confirm my theory. Either way, I will apply the fix correctly so it targets the correct versions of `ng-packagr`.

**Comment by leosvelperez** [comment]
Thanks for confirming!

> You're thinking it might be related to [ng-packagr/ng-packagr@3c27869](https://github.com/ng-packagr/ng-packagr/commit/3c27869ea499064b5362e736ff9fed9f2ed84ba9) ?

Yes. That's the fix. We already have the fix in place, but as I said, it's wrongly only applying for `ng-packagr` versions equal to or greater than 17.3.0. Since you have 17.2.1, it's not being applied. I'll amend that.

**Comment by leosvelperez** [comment]
@Squixx, the fix was merged and released in Nx 18.1.3.

**Comment by Squixx** [comment]
Yep perfect, thanks!
